### ISSUE TITLE
[Bug]: Terraform should be sending RequestTags with Create* calls

### ISSUE THREAD

**Issue by dee-kryvenko** [hashicorp/terraform-provider-aws#36515]
### Terraform Core Version

N/A

### AWS Provider Version

N/A

### Affected Resource(s)

`aws_s3_bucket`, possibly others

### Expected Behavior

This is a common practice to require certain tags via IAM policies. I.e. to use `resource: *` in the IAM but instead, use conditions, which separates create statements against RequestTags and all other statements against ResourceTags.

This provider currently doesn't respect that, it tries to first create a bucket, and then make separate request to tag it. Which means I need to allow my session to create just any bucket and then allow it to tag just any bucket, because obviously I am not having a separate policy for every individual bucket... this is not practical. How's anyone even using it at a scale? There was one similar PR that I can find https://github.com/hashicorp/terraform-provider-aws/pull/6396 but that fixes one singular resource.. I believe this needs to be a comprehensive fix.

### Actual Behavior

Requests to create resources are not sent with RequestTags

### Terraform Configuration Files

N/A

### Steps to Reproduce

N/A

### Would you like to implement a fix?

No

**Comment by justinretzolk** [comment]
Hey @dee-kryvenko 👋 Thank you for taking the time to raise this! If I'm not mistaken, this may be configuration dependent, related to whether you have tags that are marked as `(known after apply)`.  Are you able to supply a sample configuration that reproduces this behavior?

**Comment by dee-kryvenko** [comment]
Not true, you can see it in code https://github.com/hashicorp/terraform-provider-aws/blob/v5.42.0/internal/service/s3/bucket.go#L727-L760

Not at any point tags specification is added to the create request, and it's not until after the fact the tags are being set https://github.com/hashicorp/terraform-provider-aws/blob/v5.42.0/internal/service/s3/bucket.go#L776-L778. Well at least for S3, not sure about every other resource but looking at https://github.com/search?q=repo%3Ahashicorp%2Fterraform-provider-aws%20TagSpecifications&type=code I'm seeing only a handful of them does. I would imagine/hope this is code generated, would be a huge hassle to fix every single resource manually. And until it is - it cripples adoption of good security hygiene since most of the people will just use IAM policy that works, and not get out of their way to file issue on GitHub (unfortunately).

**Comment by SolomonHD** [comment]
My situation: Tagging required for volumes at ec2 instance creation. Currently there is no way for `aws_instance` to work, I've tried all variables: `tags, tags_all, volume_tags` , none actually tag at creation--although `volume_tags` implies it does.

However, I found a workaround in the [AutoScaling Group Module](https://registry.terraform.io/modules/HDE/autoscaling/aws/latest), specifically the `tag_specifications` option with the resource type set to `instance` (oddly type `volume` does not work!)

Since there's not much practical difference between an ASG of 1 and an EC2 I'll use this module until this bug is fixed.

**Comment by teads-bill-met** [comment]
This is alos the case for https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/batch_compute_environment

**Comment by brianmaresca** [comment]
is there any update on this? not including request tags for various resources including s3 has proven to be a massive pain and ultimately a security risk. we totally lose out on the ability to take advantage of the aws:RequestTag/${TagKey} iam condition key.

**Comment by blazehilario** [comment]
I am having the issue as well! is there any update on this?

**Comment by JimmyHuang0904** [comment]
Right now I have to resort to adding individual tags argument to resources I care about that needs to use aws:RequestTag/${TagKey} IAM condition. This defeats the purpose of a default_tags for me on the provider block.

**Comment by brianmaresca** [comment]
plz respond?
